Modules/_asynciomodule.c

```diff
@@ -612,25 +612,40 @@ future_set_exception(FutureObj *fut, PyObject *exc)
 }
 
 static PyObject *
-create_cancelled_error(PyObject *msg)
+create_cancelled_error(FutureObj *fut)
 {
     PyObject *exc;
+    if (fut->fut_cancelled_exc != NULL) {
+        exc = fut->fut_cancelled_exc;
+        /* Check if exc is a CancelledError */
+        int res = PyObject_IsInstance(exc, asyncio_CancelledError);
```

Comment:
In what case it is not a CancelledError?
If it can be not a CancelledError, why not use `PyObject_TypeCheck()`?

Reply:
Good catch!
The only CancelledError can be here.